registry/spec.md

```diff
@@ -1483,165 +1624,1718 @@ An attempt to delete all versions MUST generate an error.
+for network endpoints, which accept or emit events and messages defined therein.
+
+The CloudEvents registry model contains three separate registries that can be
+implemented separately or in combination.
+
+- The [Schema Registry](#schema-registry) section describes the metadata
+  description of payload schemas for events and messages. The schema registry is
+  universally applicable to any scenario where collaborating parties share
+  structured data that is defined by formal schemas. For instance, when storing
+  Protobuf encoded structured data in a cloud file store, you might place a
+  schema registry in file form in the parent directory, which formally organizes
+  and documents all versions of all Protobuf schemas that are used in the
+  directory.
+- The [Message Definitions Registry](#message-definitions-registry) section
+  describes the metadata description of events and messages. The payload schemas
+  for events and messages can be embedded in the definition, reference an
+  external schema document, or can be referenced into the schema registry. The
+  message definitions registry is universally applicable to any asynchronous
+  messaging and eventing scenario. You might define a group of definitions that
+  describe precisely which messageds with which metadata are permitted to flow
+  into a channel and can thus be expected by consumers of that channel and then
+  associate that definition group with a topic or queue in your eventing or
+  messaging infrastructure. That association might be a metadata attribute on
+  the topic or queue in the messaging infrastructure that embeds the metadata or
+  points to it.
+- The [Endpoint Registry](#endpoint-registry) section defines the metadata
+  description of network endpoints that accept or emit events and messages. The
+  endpoint registry is a formal description of associations of message
+  definitions and network endpoints, which can be used to discover endpoints
+  that consume or emit particular messages or events via a central registry. The
+  message definitions can be embedded into the endpoint metadata or as
+  a reference into the message definitions registry.
+
+The metadata model is structured such that network endpoint information and
+message metadata and payload schemas can be described compactly in a single
+object in the simplest case or can be spread out and managed across separate
+registry products in a sophisticated large-enterprise scenario.
+
+The following is an exemplary, compact definition of an MQTT 5.0 consumer
+endpoint with a single, embedded message definition using an embedded Protobuf 3
+schema for its payload.
+
+``` JSON
+{
+    "$schema": "https://cloudevents.io/schemas/registry",
+    "specversion": "0.3-wip",
+    "id": "urn:uuid:3978344f-8596-4c3a-a978-8fc9a6a469f7",
+    "endpoints" : 
+    {
+       "com.example.telemetry" : {
+           "type": "endpoint",
+           "id" : "com.example.telemetry",
+            "usage": "consumer",
```

Comment:
Missing "name" and "epoch".
The indent of "usage" (and following lines) are off by one char.

Reply:
YAML-trained eyes.

* We should discuss whether we really need `name`. It's at most an override for `id`.
* `epoch` makes sense to have in API-hosted docs, but I would like to make it optional (quasi: required, but default to zero) in files.